### ISSUE TITLE
Some Accessibility issues in Settings Screen

### ISSUE THREAD

**Issue by zhou-zhoukang** [Semper-Viventem/MetaRadar#200]
**Describe the bug**
Hello developer, I noticed there are some accessibility issues in settings screen. Taking the example of the interface shown in the figure below, the red boxes indicate the order in which components are focused sequentially by screen readers when visually impaired users navigate through the UI. When they focus on the Switch, they will only hear its current on/off state but cannot understand what specific feature this Switch is enabling/disabling. This requires them to perform additional left/right swipes to obtain contextual information about the page. This may creates unnecessary cognitive load.

A potential approach would be using Compose's semantic APIs to put the Switch and its description component together, as demonstrated in Compose's official accessibility [documentation](https://developer.android.com/develop/ui/compose/accessibility/merging-clearing). Alternatively, using `ListItem` is a more convenient approach -- just place the `Switch` in the `trailingContent` section, since `ListItem` is a component in Compose Material 3 that has already been optimized for accessibility internally for developers.

**Screenshots**
![Image](https://github.com/user-attachments/assets/2cc13b36-accc-4265-9d66-bc596948e645)

**Smartphone (please complete the following information):**
 - Not Important

**Additional context**
I completely understand if this is a lower priority because it is just a minor accessibility issue. Thank you for your hard work!
